Provider Crashes with nil pointer on Okta resource_okta_brand

Terraform Panics with a nil pointer every time I try to run `terraform apply` on the `okta_brand` resource.

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.5.7

### Affected Resource(s)

- okta_brand

### Panic Output

Stack trace from the terraform-provider-okta_v4.4.3 plugin:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x203aaa9]

goroutine 429 [running]:
github.com/okta/terraform-provider-okta/okta.(*brandResource).Read(0xc000012360, {0x26aec60, 0xc0007b1b00}, {{{{0x26b3fc0, 0xc0008a3200}, {0x2267ec0, 0xc0008a2f90}}, {0x26b5948, 0xc00058e1e0}}, 0xc000012370, ...}, ...)
	github.com/okta/terraform-provider-okta/okta/resource_okta_brand.go:220 +0x3a9
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc0000eb4a0, {0x26aec60, 0xc0007b1b00}, 0xc0007b1b90, 0xc0005e15a8)
	github.com/hashicorp/terraform-plugin-framework@v1.4.1/internal/fwserver/server_readresource.go:101 +0x677
github.com/hashicorp/terraform-plugin-framework/internal/proto5server.(*Server).ReadResource(0xc0000eb4a0, {0x26aec60?, 0xc0007b19b0?}, 0xc000404b80)
	github.com/hashicorp/terraform-plugin-framework@v1.4.1/internal/proto5server/server_readresource.go:56 +0x41a
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ReadResource(0x26aebb8?, {0x26aec60?, 0xc0007b1590?}, 0xc000404b80)
	github.com/hashicorp/terraform-plugin-mux@v0.12.0/tf5muxserver/mux_server_ReadResource.go:35 +0x1b5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc000376140, {0x26aec60?, 0xc0007b0bd0?}, 0xc00004a540)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:789 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x23f9860?, 0xc000376140}, {0x26aec60, 0xc0007b0bd0}, 0xc000288150, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:431 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00016e5a0, {0x26b42f8, 0xc000702340}, 0xc00046b440, 0xc0003ffa10, 0x30ea248, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1360 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc00016e5a0, {0x26b42f8, 0xc000702340}, 0xc00046b440, 0x0)
	google.golang.org/grpc@v1.57.0/server.go:1737 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.57.0/server.go:982 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.57.0/server.go:980 +0x18c

Error: The terraform-provider-okta_v4.4.3 plugin crashed!`
```

### Expected Behavior

Brand should be updated


### Actual Behavior

Provider crash

### Steps to Reproduce

Attempt an update on these resources:

```hcl
resource "okta_brand" "default_brand" {
  brand_id               = "default"
  name                   = "brand_default"
  remove_powered_by_okta = true
}

resource "okta_brand" "custom_brand" {
  name                   = "brand"
  remove_powered_by_okta = false
}
```

--- a/okta/resource_okta_brand.go
+++ b/okta/resource_okta_brand.go
@@ -215,10 +215,13 @@ func (r *brandResource) Read(ctx context.Context, req resource.ReadRequest, resp
 
 	if state.BrandID.ValueString() == "default" {
 		brands, _, err := r.oktaSDKClientV3.CustomizationApi.ListBrands(ctx).Execute()
-		resp.Diagnostics.AddError(
-			"failed to get list brand",
-			err.Error(),
-		)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"failed to get list brand",
+				err.Error(),
+			)
+			return
+		}
 
 		for _, brand := range brands {
 			if *brand.IsDefault {

